### ISSUE TITLE
Breakup Long Functions

### ISSUE THREAD

**Issue by hbreauxv** [hbreauxv/NCM-Bin-Upload#34]
Some of the feedback from js dev friends of mine was to break up some of the longer functions into smaller functions to improve readability and keep separate tasks separated in code

**Comment by hbreauxv** [comment]
Specific feedback for posterity:

`Array.prototype.slice.call(divTags)` is a bit odd, you can just use `Array.prototype.slice(divTags)` (or use Array.from(divTags))

The findParent (and also findChild) function can be rewritten:

`async function findParent(time) {
   const divTags = document.getElementsByTagName("div");
   const parent = Array.from(divTags).find(tag =>
       divTag.getAttribute("id").includes("ecm-core-view-devices-Routers") && divTag.getAttribute("id").includes("body") === false
   )
   return parent !== undefined ? parent : await new Promise((resolve, reject) => setTimeout(resolve(findParent(time)), time))
}`

Basically taking advantage of the .find method to save looping and then a recursive call. 
(Also `tag => ...stuff` is just a shorter version of `function(tag) {stuff}` and has some benefits when it comes to the `this` keyword.

**Comment by hbreauxv** [comment]
That formatting is awful, which is more proof that google messages is a terrible place to send code snippets :)

**Comment by hbreauxv** [comment]
Resolved by #49
